Automattic/wp-calypso
# Checkout: fix line item display coupon and discount display

The new Checkout's endpoint translation methods have a temporary fix for displaying localized line items and coupon discounts. This replaces those methods with properly localized values returned from the server. 

This additionally adds support for domains in the total cart savings (coupon) line item. Previously, this was calculated based on mapped and reduced values from the `coupon_discounts_` array from the endpoint, which only contains discounts applied to plans.

Design updates to the line items will come in a future PR.

See D40360-code
Fixes: #39990
Fixes: #40218

![Screen Shot 2020-04-23 at 11 54 39 PM](https://user-images.githubusercontent.com/942359/80173484-06a55380-85be-11ea-95b2-89af3dca551c.png)

**To test:**
- Apply D40360-code and sandbox `public-api.wordpress.com`
- Visit Checkout with a domain and a plan in your cart
- Verify that the line item values are rounded to the appropriate currency decimal place
- Apply a coupon to the cart
- Verify that the correct original cost for each product is now crossed out, with the correct discounted price beside it
- Verify that the coupon line item is the correct sum of all coupon discounts applied to the cart
- Verify that the cart totals and taxes are correctly calculated
- Verify that you can complete checkout and purchase the items
